src/task.c

```diff
@@ -91,10 +91,11 @@ task_init_threadid(task_system *ts)
    }
 
 out:
+   // The main program counts for thread-0; hence limit is "-1"
    platform_assert((tid != INVALID_TID),
                    "Cannot create a new thread as the limit on"
                    " concurrent threads, %d, will be exceeded.\n",
-                   MAX_THREADS);
+                   (MAX_THREADS - 1));
```

Comment:
Slightly unrelated fix. 

Without this fix, we will be reporting an error like so, when  `--num-insert-threads 64` is used:

```
Assertion failed at src/task.c:95:task_init_threadid(): "(tid != INVALID_TID)". Cannot create a new thread as the limit on concurrent threads, 64, will be exceeded.
```

The real-limit (externally) is 63 (user + bg-threads)